### ISSUE TITLE
1.17.1 auto versioning no longer working

### ISSUE THREAD

**Issue by CyanCode** [PrismarineJS/node-minecraft-protocol#1086]
[✅ ] The [FAQ](https://github.com/PrismarineJS/node-minecraft-protocol/blob/master/docs/FAQ.md) doesn't contain a resolution to my issue 

## Versions
 - minecraft-protocol: 1.36.0 and up
 - server: vanilla
 - node: 18.4.0

## Detailed description of a problem
Attempting to login on Java vanilla 1.17.1 client fails when using auto version. This started happening after upgrading to `1.41.0` so I checked the versions between `1.26.5` and `1.41.0`. [Version 1.36.0](https://github.com/PrismarineJS/node-minecraft-protocol/releases/tag/1.36.0) is where this starts happening and is likely the source of the regression.

```
PartialReadError: Read error for undefined : undefined
    at new ExtendableError (C:\Users\walke\Desktop\Folders\Projects\mc-apps\api\node_modules\protodef\src\utils.js:63:13)
    at new PartialReadError (C:\Users\walke\Desktop\Folders\Projects\mc-apps\api\node_modules\protodef\src\utils.js:70:5)
    at Object.readBool [as bool] (C:\Users\walke\Desktop\Folders\Projects\mc-apps\api\node_modules\protodef\src\datatypes\utils.js:133:41)
    at eval (eval at compile (C:\Users\walke\Desktop\Folders\Projects\mc-apps\api\node_modules\protodef\src\compiler.js:258:12), <anonymous>:611:29)
    at Object.packet_login_start (eval at compile (C:\Users\walke\Desktop\Folders\Projects\mc-apps\api\node_modules\protodef\src\compiler.js:258:12), <anonymous>:617:9)
    at eval (eval at compile (C:\Users\walke\Desktop\Folders\Projects\mc-apps\api\node_modules\protodef\src\compiler.js:258:12), <anonymous>:658:62)
    at packet (eval at compile (C:\Users\walke\Desktop\Folders\Projects\mc-apps\api\node_modules\protodef\src\compiler.js:258:12), <anonymous>:663:9)
    at CompiledProtodef.read (C:\Users\walke\Desktop\Folders\Projects\mc-apps\api\node_modules\protodef\src\compiler.js:70:12)
    at e.message (C:\Users\walke\Desktop\Folders\Projects\mc-apps\api\node_modules\protodef\src\compiler.js:111:49)
    at tryCatch (C:\Users\walke\Desktop\Folders\Projects\mc-apps\api\node_modules\protodef\src\utils.js:50:16)
```

## Current code
```js
      const address = await getLocalAddress();
      const mcServer = mc.createServer({
        'online-mode': false,
        host: address,
        port: MC_SERVER_PORT,
        keepAlive: true,
      });

      mcServer.on('error', (e) => {
        this.logger.error('Error occurred while listening for logins', e);
      });
      mcServer.on(
        'login',
        async (client: ServerClient) => await this.onClientLogin(client),
      );
      mcServer.on('listening', () => {
        this.logger.info(
          `Listening for login attempts on ${address}:${MC_SERVER_PORT}`,
        );
        resolve();
      });
```

## Expected behavior
Auto version works as it had previously allowing incoming 1.17.1 connections to be correctly resolved

## Additional context
Looks like a similar issue was opened for this a while ago https://github.com/PrismarineJS/node-minecraft-protocol/issues/928

**Comment by extremeheat** [comment]
Known issue in latest release, temporarily fixed with #1084

**Comment by CyanCode** [comment]
Gotcha, thanks for the update! I'll wait for the next release.
